### PR TITLE
SelectionStream: Use ctor instead of C++11 init

### PR DESCRIPTION
**Pull request by fritsch**
If we would not do that - we would break Ubuntu 12.04.

### CODE REVIEW

**Comment by fritsch** [comment]
@ace20022 can you have a look please - I only did a technical fix here, but did not check all the ways it is involved.

**Comment by fritsch** [comment]
@wsnipex could you trigger another build? So we can fix up the rest if possible?

File: xbmc/cores/dvdplayer/DVDPlayer.h [hunk 1]
```diff
@@ -154,17 +154,26 @@ class CCurrentStream
...
-  CDemuxStream::EFlags flags = CDemuxStream::FLAG_NONE;
-  int          source = 0;
-  int          id = 0;
+  CDemuxStream::EFlags flags;
+  int          source;
+  int          id;
   std::string  codec;
-  int          channels = 0;
+  int          channels;
+  SelectionStream()
```
Comment (Montellese) [inline xbmc/cores/dvdplayer/DVDPlayer.h]: insert an empty line for better readability.

Comment (fritsch) [inline xbmc/cores/dvdplayer/DVDPlayer.h]: Will do, thx.

**Comment by fritsch** [comment]
Let's see if our jenkins likes it and then wait for @wsnipex and the ppa builder.

jenkins build this please

**Comment by wsnipex** [comment]
https://launchpadlibrarian.net/211969752/buildlog_ubuntu-precise-i386.kodi_2%3A16.0~git20150718.1502-a2523a6-0precise_BUILDING.txt.gz

**Comment by Paxxi** [comment]
@fritsch small thing, if you don't mind, use the initializer list instead of constructor body, it'll give cppcheck perf warnings this way. Will probably get a patch for it later on once someone's had a go with it :) doesn't affect functionality so take it as a low prio suggestion.

**Comment by fritsch** [comment]
0xb6d4c81c forward declaration for enum classes :(

**Comment by fritsch** [comment]
From my pov it makes no sense to rework the enum classes to something old-school. What do the others think?

**Comment by wsnipex** [comment]
I'm fine with dropping precise for PPA builds, users can still build themselves if they upgrade gcc

**Comment by arnova** [comment]
IMHO we might as well use the init-list. It's not like we don't use it in other places ;-)

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
@MartijnKaijser this can probably be closed as it didn't fix the <=gcc4.6 issue and we decided to drop ubuntu 12.04 support from the PPA

**Pull request closed without merging**